iFixit/react-commerce
# Upgrade ts to 5.2.2

This PR upgrade TS to 5.2.2.
The reason for this is that:
> To use an async Server Component with TypeScript, ensure you are using TypeScript 5.1.3 or higher and @types/react 18.2.8 or higher.

However, Chakra typings seems to be not 100% compliant with ts5.
See [here](https://github.com/chakra-ui/chakra-ui/pull/7506) and [here](https://github.com/chakra-ui/chakra-ui/issues/7459). The only working combination of chakra, react, react-dom etc. packages that I found was not producing compilation errors or a bunch of IDE errors is the one in this PR, which uses @types/react 18.2.0.

Moreover, we had to upgrade Chakra, and even though no new major was released, a [breaking change was shipped](https://github.com/chakra-ui/chakra-ui/issues/7269#issuecomment-1424868672) so also this was taken into account in this PR.

### QA
1. Visit vercel preview
2. Verify that nothing visible was changed in the ecommerce ui (Troubleshooting, Product, Productlist)
3. Verify that on local dev environment no new errors wrt the past are shown
